test/remote/gateways/remote_priority_test.rb

```diff
@@ -241,7 +238,7 @@ def test_successful_refund_and_batch_closed
       @gateway.close_batch(response.params['batchId'], @option_spr)
       refund_params = @option_spr.merge(response.params).deep_transform_keys { |key| key.to_s.underscore }.transform_keys(&:to_sym)
 
-      refund = @gateway.refund(response.params['amount'].to_f * 100, @credit_card, refund_params)
+      refund = @gateway.refund(response.params['amount'].to_f * 100, response.authorization.to_s, refund_params)
```

Comment:
I think this change is fine for the context of this PR. I was able to verify that it is working.

However, I think it's kind of bad news that this test file contains a few tests with `if/else` statements. It means that when I run the tests I am not hitting all of the assertions. If I run the tests and they follow Path A, who's to say that Path B works?

Case in point: I had to put a breakpoint in this test just to verify that I was actually hitting this code when I run the test. Someone could make a bunch of changes in this `if` block and keep running the test and keep getting a green result without realizing that the test isn't even hitting their code.

I think it would be worthwhile to generate a separate task to rethink some of these tests. I don't think we want tests with multiple paths to green, and I wouldn't want people to replicate this pattern for other gateways.

Reply:
Circling back around cause I just realized something: I think maybe the original intention of these `else` blocks was to make an assertion that is guaranteed to fail? The comments next to these tests (`# if batch Open then fail test`) seems to indicate this intention, although the comments are inaccurate (if the batch is open we go ahead and close it and proceed as planned).

Making an `assert` statement that is bound to fail is not a good way to communicate this intention. The presence of an `if/else` further confuses things, as it suggests there are multiple acceptable paths the test could follow.

RSpec offers a `flunk` method for just this purpose, and if we wrap it in a guard clause I think the intention becomes very clear:
```
flunk('the batch status prevents a refund from being tested') unless batch_check.params['status'] == 'Open'
```

Reply:
Thanks @dsmcclain ! I've created a Jira ticket (CE-2253) to track these updates 👍🏻